cerberus_anymal_control_common/cerberus_anymal_control_base/include/cerberus_anymal_control_base/CerberusAnymalControlBase.tpp

```diff
@@ -0,0 +1,70 @@
+/*!
+ * @name      CerberusAnymalControlBase.tpp
+ * @author    Samuel Zimmermann
+ * @email     zsamuel@ethz.ch
+ * @date      April 30, 2020
+ *
+ * Copyright (C) 2020 Robotic Systems Lab, ETH Zurich.
+ * All rights reserved.
+ * http://www.rsl.ethz.ch/
+ */
+
+#include "CerberusAnymalControlBase.hpp"
+
+namespace cerberus_anymal_control {
+
+template <typename ConcreteQuadrupedController>
+CerberusAnymalControlBase<ConcreteQuadrupedController>::CerberusAnymalControlBase(int numberOfJoints) :
+    quadrupedController_(),
+    graph_(),
+    numberOfJoints_(numberOfJoints),
+    genCoordinates_(),
+    genVelocities_(),
+    desLinearVelocity_(Eigen::Vector3d::Zero()),
+    desAngularVelocity_(Eigen::Vector3d::Zero()),
+    desJointPositions_(Eigen::VectorXd::Zero(numberOfJoints)),
+    hasBodyVelocityData_(false),
+    hasJointPositionData_(false),
+    initializedGraph_(false)
+{
+  quadrupedController_ = std::make_unique<ConcreteQuadrupedController>();
+  if (numberOfJoints_ != 12) {
+    std::cout << "[CerberusAnymalControlBase::CerberusAnymalControlBase] Joint Number is not equal to 12! Controller might not be working properly." << std::endl;
+  }
+  // Fix the pose since we already subscribe in base frame
+  genCoordinates_(0) = 0.0;
```

Comment:
is there a block operator to make this assignment in one line? 7 Duplicated lines of code made my hearth skipping 7 beats